pico/vga.c

```diff
@@ -146,12 +143,11 @@ static void vga_dma_irq_handler() {
     active_scanline->_flags &= ~(uint_fast8_t)FLAG_BUSY;
 
     const uint32_t irq_status = spin_lock_blocking(lock);
-    scanline_queue_tail = (scanline_queue_tail + 1) % NUM_SCANLINE_BUFFERS;
+    scanline_queue_tail = (scanline_queue_tail + 1) % (NUM_SCANLINE_BUFFERS - 1);
```

Comment:
I don't think this should be necessary and will just waste a buffer, won't it?  Did this solve an issue or was it just copy/pasted from the analog-firmware code?

Reply:
reverted to your code, When I was working on the video modes, it was not working due to other reasons, so I thought maybe it has to do with some things, so I experimented with the other source, but now that I understand why it was not working, this can be reverted.